lib/activerecord-bitemporal.rb

```diff
@@ -70,6 +70,15 @@ def load_schema!
   module InstanceMethods
     include ActiveRecord::Bitemporal::Persistence
 
+    def initialize_dup(other)
+      # MEMO: Do not copy `swapped_id`
+      if instance_variable_defined?(:@_swapped_id)
+        remove_instance_variable(:@_swapped_id)
+      end
+
+      super
+    end
```

Comment:
Doesn't it break when `#initialize_dup` is defined in user code?

```ruby
# User code
class User < ApplicationRecord
  include

  def initialize_dup
    # other implemented
  end
end
```

Reply:
Also, doesn't this implementation also change the behavior of code that calls `#dup` elsewhere?

Reply:
> Doesn't it break when #initialize_dup is defined in user code?

It isn't problem because it is the same as `#dup`. Normally user shouldn't define `initialize_dup` without super.

> Also, doesn't this implementation also change the behavior of code that calls #dup elsewhere?

Right. I just thought that `#dup` should clear instance scoped variables but it is not necessary as you said.
I just resolved this issue by replacing `refine Object` with `refine ActiveRecord::Base`.